src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -561,3 +562,86 @@ class SetStatementSegment(BaseSegment):
             )
         ),
     )
+
+
+@bigquery_dialect.segment()
+class PartitionBySegment(BaseSegment):
+    """PARTITION BY partition_expression."""
+
+    type = "partition_by_segment"
+    match_grammar = Sequence(
+        "PARTITION",
+        "BY",
+        Delimited(Ref("ExpressionSegment")),
+    )
+
+
+@bigquery_dialect.segment()
+class ClusterBySegment(BaseSegment):
+    """CLUSTER BY clustering_column_list."""
+
+    type = "cluster_by_segment"
+    # PARTITION BY partition_expression
+    match_grammar = Sequence(
+        "CLUSTER",
+        "BY",
+        Delimited(Ref("ExpressionSegment")),
+    )
+
+
+@bigquery_dialect.segment()
+class OptionsSegment(BaseSegment):
+    """OPTIONS clause for a table."""
+
+    type = "options_segment"
+    match_grammar = Sequence(
+        "OPTIONS",
+        Bracketed(
+            Delimited(
+                # Table options
+                Sequence(
+                    Ref("ParameterNameSegment"),
+                    Ref("EqualsSegment"),
+                    Ref("LiteralGrammar"),
+                )
+            )
+        ),
+    )
+
+
+@bigquery_dialect.segment(replace=True)
+class CreateTableStatementSegment(BaseSegment):
+    """A `CREATE TABLE` statement."""
+
+    type = "create_table_statement"
+    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_table_statement
```

Comment:
In addition to the "OPTIONS" clause, BigQuery "CREATE TABLE" also has "PARTITION BY" and "CLUSTER BY". Together, these differences seem big enough to justify replacing `CreateTableStatementSegment` rather than just customizing/extending it. That's why I went ahead and added support for "PARTITION BY" and "CLUSTER BY" even though they aren't part of issue #1205.